src/parsing.ts

```diff
@@ -87,31 +94,18 @@ export const learnFileId = async (_graph: Graph, filePath: string) => {
 
 export const parseDirectory = async (
   graph: Graph,
-  directory: string,
   fileCallback: (graph: Graph, path: string) => Promise<void>
 ) => {
-  const files = await vscode.workspace.fs.readDirectory(
-    vscode.Uri.file(directory)
+  const files = await vscode.workspace.findFiles(
+    `**/*{${(getFileTypesSetting() as string[]).map((f) => `.${f}`).join(",")}}`
   );
 
   const promises: Promise<void>[] = [];
 
   for (const file of files) {
-    const fileName = file[0];
-    const fileType = file[1];
-    const isDirectory = fileType === vscode.FileType.Directory;
-    const isFile = fileType === vscode.FileType.File;
-    const hiddenFile = fileName.startsWith(".");
-    const isGraphFile = getFileTypesSetting().includes(
-      fileName.substr(fileName.lastIndexOf('.') + 1)
-    )
-
-    if (isDirectory && !hiddenFile) {
-      promises.push(
-        parseDirectory(graph, `${directory}/${fileName}`, fileCallback)
-      );
```

Comment:
I assume that `vscode.workspace.findFiles` returns all files and there's no need to recursively look for directories, right?

Can you check if [functionality](https://github.com/tchayen/markdown-links/tree/master/examples/functionality) example supports `/nested` files too?

Reply:
Hmmm, I suspect these are meant to be at least a little bit linked, right?

![image](https://user-images.githubusercontent.com/22981941/88383752-0118fe80-cda3-11ea-82fc-f2ea480052ab.png)

Reply:
It's showing them at the very least!

Reply:
Hmmm, 99% sure they were 😄

Reply:
@tchayen Okay, I'll take a look...

- [ ] Get nested links linking again

Reply:
@tchayen Actually...would you mind giving this a look? I've just double checked on the current released version of the extension and it seems there's no links there either...

Reply:
Looks like #26 could be interfering with my results, as I'm on a windows machine

Reply:
Ok, I am going to check it now 👀 .

Reply:
@tchayen 
![image](https://user-images.githubusercontent.com/22981941/88466505-cdce9080-cec4-11ea-83d8-c43ad241d1fe.png)
Here's the result on MacOS. Does this make it good to merge? (seeing as it doesn't look like I've broken anything 😅)